op-dispute-mon/mon/monitor_test.go

```diff
@@ -0,0 +1,248 @@
+package mon
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/types"
+	"github.com/ethereum-optimism/optimism/op-service/clock"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMonitor_MinGameTimestamp(t *testing.T) {
+	t.Parallel()
+
+	t.Run("zero game window returns zero", func(t *testing.T) {
+		monitor, _, _, _ := setupMonitorTest(t)
+		monitor.gameWindow = time.Duration(0)
+		require.Equal(t, monitor.minGameTimestamp(), uint64(0))
+	})
+
+	t.Run("non-zero game window with zero clock", func(t *testing.T) {
+		monitor, _, _, _ := setupMonitorTest(t)
+		monitor.gameWindow = time.Minute
+		monitor.clock = clock.NewSimpleClock()
+		monitor.clock.SetTime(0)
+		require.Equal(t, uint64(0), monitor.minGameTimestamp())
+	})
+
+	t.Run("minimum computed correctly", func(t *testing.T) {
+		monitor, _, _, _ := setupMonitorTest(t)
+		monitor.gameWindow = time.Minute
+		monitor.clock = clock.NewSimpleClock()
+		frozen := uint64(time.Hour.Seconds())
+		monitor.clock.SetTime(frozen)
+		expected := uint64(time.Unix(int64(frozen), 0).Add(-time.Minute).Unix())
+		require.Equal(t, monitor.minGameTimestamp(), expected)
+	})
+}
+
+func TestMonitor_RecordGameStatus(t *testing.T) {
```

Comment:
We should setup these tests to have multiple games in different types and assert that the counts are added up correctly.  Because they're always working with a single game, and working at the `recordGameStatus` level rather than `monitorGames`, they're missing the bug that only the last game is ever recorded in metrics.

Reply:
I've refactored this out in the pr stacked ontop. If it's ok with you, I'd like to clean up the tests in https://github.com/ethereum-optimism/optimism/pull/9422 since it nicely refactors code out of the monitor which should ideally in my mind only be responsible for the scheduling / ticking logic around dispatching "detector" checks.

If comprehensive testing in this pr is preferred, I can do that, it may just take more time and expand this pr further.